Fix rails 5

jspizziri opened the pull request:
Rails 5 will error out because the interfaces are not being required and autoloading is disabled.

jspizziri committed:
fix(spree_chimpy, order_upserter, orders): Fix issue with no autoloading in Rails 5 breaking gem

jspizziri committed:
fix(spree_chimpy): Fix lazyload files since Rails.application is not loaded yet

mrogers-5s committed:
fix(lib/spree/chimpy/interface/products.rb): Fix MailChimp tracking integrations

Modify the self.variant_hash method of the Products interface
-- Only include the :image_url if a
vairant image is present. This is an optional key, but causes an error in the MailChimp API if set
to nil. It should be excluded if it's not defined on the object.

Fix bug in MailChimp Campaign tracking
-- Store MailChimp params in session hash until they can be attached to an order

mrogers-5s committed:
feat(carts): Add support for tracking Carts in MailChimp
Add integration with the Carts endpoint in the MailChimp
API to allow the use of Abandoned Cart automations

feat(cart tracking): Add support for tracking Carts in MailChimp

-- Add support for tracking carts in MailChimp to allow the use of Abandoned Cart campaigns.

mrogers-5s committed:
fix(lib/spree/chimpy/interface/products.rb): Fix bug POST call for new products when variants have n

Modify the self.variant_hash method of the Products interface
-- Only include the :image_url if a
vairant image is present. This is an optional key, but causes an error in the MailChimp API if set
to nil. It should be excluded if it's not defined on the object.

mrogers-5s committed:
Fix/campaign tracking (#1)

* fix(Campaign Tracking): Fix campaign revenue tracking

Fix a bug that prevents Orders from being attached to a MailChimp campaign
-- To fix: Store
MailChimp tracking parameters in the session hash until they can be attached to an order

* chore

Remove testing comments

* fix(MC Integration): Fix campaign tracking for MC integration

-- Relocate the monkey patch for the Orders Controller helper (added to the populate_order_source
method in the model class instead.
-- Modify the Order's state machine to call new method
before_transition to any state

TODO: process still relies on the session variable, which appears
to be unavailable in the model classes. Need to resolve how to pass those MC session values into the
model

* fix(Campaign Tracking): Relocate campaign tracking to minimize core changes

Minimize changes to core Spree classes/methods
-- Move the process for populating the order source
to its own method in the OrdersController and call this method on before_action events

* fix(Campaign Tracking): Fix bug that prevents retention of MailChimp parameters

Fix big in `controller_filters.rb` which prevents MailChimp tracking parameters from being properly
stored/attached to an order

* chore(Products Interface) Whitespace changes

Add whitespace for readability

* chore(ControllerFilters) Fix typo in comments

* chore(ControllerFilters) fix more typos

jspizziri committed:
Merge pull request #1 from mrogers-5s/fix/campaign_tracking

Fix/campaign tracking

mrogers-5s committed:
Merge branch 'fix-rails-5' into feature/abandoned_cart_tracking

jspizziri committed:
Merge pull request #2 from mrogers-5s/feature/abandoned_cart_tracking

Feature/abandoned cart tracking

mrogers-5s committed:
fix(spree_chimpy.rb): Add missing require calls for Carts interface

Add missing call to require rb file for the Carts interface

mrogers-5s committed:
Merge branch 'feature/abandoned_cart_tracking' of github.com:mrogers-5s/spree_chimpy into feature/abandoned_cart_tracking

jmeyers44 commented:
@jspizziri do you have plans to merge this PR? Will spree_chimpy work in rails 5 without this PR?

mrogers-5s committed:
fix(carts/orders): Fix sync of carts/orders

Fix the sync process for Carts and Orders so both data elements sync correctly with MailChimp.
--
Add before_action calls in the `checkouts_controller.rb` so carts update appropriately at all
stages
-- change orders_controller to call `order.notify_mail_chimp` as `after_action` (this
ensures that during checkout completion, an order is synced AFTER the checkout is complete rather
than after a "complete order" request is submitted

jspizziri committed:
Merge pull request #3 from mrogers-5s/feature/abandoned_cart_tracking

Feature/abandoned cart tracking

mrogers-5s committed:
Feature/abandoned cart tracking (#4)

* fix(spree_chimpy, order_upserter, orders): Fix issue with no autoloading in Rails 5 breaking gem

* fix(spree_chimpy): Fix lazyload files since Rails.application is not loaded yet

* fix(lib/spree/chimpy/interface/products.rb): Fix MailChimp tracking integrations

Modify the self.variant_hash method of the Products interface
-- Only include the :image_url if a
vairant image is present. This is an optional key, but causes an error in the MailChimp API if set
to nil. It should be excluded if it's not defined on the object.

Fix bug in MailChimp Campaign tracking
-- Store MailChimp params in session hash until they can be attached to an order

* feat(carts): Add support for tracking Carts in MailChimp
Add integration with the Carts endpoint in the MailChimp
API to allow the use of Abandoned Cart automations

feat(cart tracking): Add support for tracking Carts in MailChimp

-- Add support for tracking carts in MailChimp to allow the use of Abandoned Cart campaigns.

* fix(spree_chimpy.rb): Add missing require calls for Carts interface

Add missing call to require rb file for the Carts interface

* fix(carts/orders): Fix sync of carts/orders

Fix the sync process for Carts and Orders so both data elements sync correctly with MailChimp.
--
Add before_action calls in the `checkouts_controller.rb` so carts update appropriately at all
stages
-- change orders_controller to call `order.notify_mail_chimp` as `after_action` (this
ensures that during checkout completion, an order is synced AFTER the checkout is complete rather
than after a "complete order" request is submitted

mrogers-5s committed:
fix(carts_upserter): Fix cart recovery URL

Fix cart recovery URL so users are directed to a page that will allow them to checkout

mrogers-5s committed:
fix(carts_upserter): Fix cart recovery URL (#5)

Fix cart recovery URL so users are directed to a page that will allow them to checkout

mrogers-5s committed:
Merge branch 'fix-rails-5' into fix/181/cart_recovery_link

jspizziri committed:
Merge pull request #4 from mrogers-5s/fix/181/cart_recovery_link

Fix/181/cart recovery link

jspizziri commented:
@jmeyers44 ,

Eventually it would be good to merge it. Although we've been a little messy with the fork and have added a bunch of other functionality (cart abandonment). Yes, in Rails 5 there are issues with the current gem that will result in production issues. I'd like to get it merged eventually; however, it looks like the upgrade broke a couple things with the travis tests, so those need to be cleaned up prior to merge.

If you're interested please feel free to clean up the tests and I'll accept the MR on my fork. That would help it get merged in faster.

jspizziri commented:
Also @bbonislawski can you please enable issue tracking on this repo?

mrogers-5s committed:
fix(cart_upserter.rb): Fix sync errors for removed cart items

Fix Cart sync process so that line items are deleted from MailChimp when they are removed from the
cart in Spree

jspizziri committed:
Merge pull request #5 from mrogers-5s/fix/183/improve_abandoned_cart_emails

Fix/183/improve abandoned cart emails

mrogers-5s committed:
fix(cart_upserter): fix uninitialized constant error

Add `require_relative` for `spree_order_upserter` interface to fix uninitialized constant errors in
production (due to disabled auto-loading)

jspizziri committed:
Merge pull request #6 from mrogers-5s/fix/169/2/cart_integration

Fix/169/2/cart integration

mrogers-5s committed:
fix(order_upserter): fix uninitialized constant

Add `require_relative` for `spree_order_upserter` in `order_upserter` to fix issues related to
disabled autoloading in procution

jspizziri committed:
Merge pull request #7 from mrogers-5s/fix/169/2/cart_integration

Fix/169/2/cart integration

molenchuk committed:
fix(lib/spree/chimpy/interface): Allow for new products and variants to be upserted

This change fixes the error where new products and variants were not being upserted to MailChimp
becasue of t a lack of ID value

jspizziri committed:
Merge pull request #8 from molenchuk/fix-products-upsert

Fix products upsert
